application/src/main/java/org/opentripplanner/routing/algorithm/RoutingWorker.java

```diff
@@ -88,19 +88,12 @@ public RoutingWorker(OtpServerRequestContext serverContext, RouteRequest request
         request.journey().transit().priorityGroupsByAgency(),
         request.journey().transit().priorityGroupsGlobal()
       );
+    this.pageCursorInputBuilder = DefaultPageCursorInput.of();
   }
 
   public RoutingResponse route() {
     OTPRequestTimeoutException.checkForTimeout();
 
-    // If no direct mode is set, then we set one.
-    // See {@link FilterTransitWhenDirectModeIsEmpty}
-    var emptyDirectModeHandler = new FilterTransitWhenDirectModeIsEmpty(
-      request.journey().direct().mode()
-    );
-
-    request.journey().direct().setMode(emptyDirectModeHandler.resolveDirectMode());
```

Comment:
Was this only used when paging or also when user decided to search with `transitOnly: true`? I don't know if we should change that behaviour.

Reply:
with `transitOnly: true` the mode gets unset here

https://github.com/opentripplanner/OpenTripPlanner/blob/7d9027691ac6446f3e97d236e03d1d078d6cfe37/application/src/main/java/org/opentripplanner/apis/gtfs/mapping/routerequest/ModePreferencesMapper.java#L29

so to me it seems that this is not related to `transitOnly`

Reply:
That code you linked sets the direct mode as NOT_SET before this code is run and this part of code that you removed probably before set the direct mode as WALK when the value was NOT_SET. So I think this part of code probably was also relevant for other purposes than paging.

Reply:
I added the emptyDirectModeHandler back and added a case for paging which makes it so that the walk mode filtering is not done with paging